api/src/middleware/team.js

```diff
@@ -23,6 +24,45 @@ async function validateTeamId(req, res, next) {
   next();
 }
 
+async function validateTeamRepo(req, res, next) {
+  const { ghToken } = req.body;
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${ghToken}` },
+    });
+  if (data.errors) {
+    next(
+      fillErrorObject(400, 'Validation error: user doesnt exist!', [data.errors[0].detail]),
+    );
+  }
```

Comment:
Pull this up into its own function in this middleware since it's being used somewhere else. See related comment: https://github.com/Researchify/Researchify/pull/198/files#r711673608

Reply:
The same as before it just retrieves the username from the token and here it exits if it is not valid!